tests/test_reinstall.py

```diff
@@ -54,3 +55,20 @@ def test_reinstall_specifier(pipx_temp_env, capsys):
     assert not run_pipx_cli(["reinstall", "--python", sys.executable, "pylint"])
     captured = capsys.readouterr()
     assert "installed package pylint 2.3.1" in captured.out
+
+
+linux_paths = ["/opt/path/file", "rel/path/", "~/tilde-path"]
+windows_paths = ["C:\\Program Files\\PATH", "D:\\TEST\\DIR", "REL\\PATH"]
+
+
+@pytest.mark.parametrize("path", windows_paths if sys.platform.startswith("win") else linux_paths)
+def test_reinstall_with_path(pipx_temp_env, capsys, path):
+    try:
+        Path(path).mkdir(parents=True, exist_ok=True)
+    except PermissionError:
+        pass
+
+    assert run_pipx_cli(["reinstall", path])
+    captured = capsys.readouterr()
+
+    assert "given as package name" in captured.err
```

Comment:
Can you write this test in a way that does not pollute the runners system?
For example for the absolute paths, use paths within the `tmp_path` of the test, and use `monkeypatch.chdir` to use relative paths within the `tmp_path`.

Reply:
Ah, right, thanks for catching.